File: CS50x/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

# Counting letters
letter = sum(c.isalpha() for c in text)

# Counting words
word = len(text.split())

# Counting sentences
sentence = text.count('.') + text.count('!') + text.count('?')

L = (letter / word) * 100
S = (sentence / word) * 100
index = (0.0588 * L) - (0.296 * S) - 15.8

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index:.0f}")

File: CS50x/readability/readability.c
#include <cs50.h>
#include <stdio.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sente = count_sentences(text);

    double L = ((double) letters / words) * 100;
    double S = ((double) sente / words) * 100;

    double index = (0.0588 * L) - (0.296 * S) - 15.8;

    if (index >= 16)
        printf("Grade 16+\n");
    else if (index < 1)
        printf("Before Grade 1\n");
    else
        printf("Grade %.0lf\n", index);

    return 0;
}

int count_letters(string text)
{
    int count = 0;
    char c;
    for (int i = 0; text[i] != '\0'; i++)

    {
        c = text[i];
        if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'))
            count++;
    }
    return count;
}

int count_words(string text)
{

    int count = 1; // for counting first word

    for (int i = 0; text[i] != '\0'; i++)

    {
        if (text[i] == ' ')
            count++;
    }
    return count;
}

int count_sentences(string text)
{

    int count = 0;
    char c;

    for (int i = 0; text[i] != '\0'; i++)

    {
        c = text[i];
        if (c == '.' || c == '!' || c == '?')
            count++;
    }
    return count;
}
